### PR TITLE
Open-API: `AssertRefSnapshotId` type should be `branch` or `tag`

### PR DESCRIPTION
**Pull request by Fokko**
I found this while reviewing a PyIceberg PR.

### CODE REVIEW

**Review by flyrain — APPROVED** [review]
Thanks for the fix. Looks good to me. It was OK since Java code doesn't use the field `type`.
```
  class AssertRefSnapshotID implements UpdateRequirement {
    private final String name;
    private final Long snapshotId;

    public AssertRefSnapshotID(String name, Long snapshotId) {
      this.name = name;
      this.snapshotId = snapshotId;
    }
```

**Comment by ajantha-bhat** [comment]
@flyrain, @Fokko: We should also update the  `AssertRefSnapshotID` to validate the `type` passed and compare against the fetched reference type using name? (as `type` is a required field)

File: open-api/rest-catalog-open-api.yaml [hunk 1]
```diff
@@ -2643,7 +2643,7 @@ components:
       properties:
         type:
           type: string
-          enum: [ "assert-ref-snapshot-id" ]
+          enum: ["tag", "branch"]
```
Comment (nastra) [inline open-api/rest-catalog-open-api.yaml:2646]: I don't think this change is correct. In https://github.com/apache/iceberg/blob/d69ba0568a2e07dfb5af233350ad5668d9aef134/core/src/main/java/org/apache/iceberg/UpdateRequirementParser.java#L169 we're relying on the `type` being `assert-ref-snapshot-id`

**Review by nastra — CHANGES_REQUESTED** [review]
I don't think this change is correct. In `UpdateRequirementParser` we're relying on the `type` being `assert-ref-snapshot-id`

**Comment by Fokko** [comment]
Hehe, you're right @nastra, thanks! The type is not specified. What confused me is that `SnapshotReference` has a type enum with `[branch, tag]`.

File: open-api/rest-catalog-open-api.yaml (hunk 1 above)
Comment (flyrain) [reply open-api/rest-catalog-open-api.yaml:2646]: I think the type duplicates the one in the parent entity. @anuragmantri made a fix here, https://github.com/apache/iceberg/pull/10434

**Pull request closed without merging**